plugins/protocols/session-lock.cpp

```diff
@@ -134,20 +134,30 @@ class lock_crashed_node : public lock_base_node<simple_text_node_t>
         set_size(output->get_screen_size());
     }
 
-    void display()
+    void display(std::string text)
     {
         wf::cairo_text_t::params params(
             1280 /* font_size */,
             wf::color_t{0, 0, 0, 1} /* bg_color */,
             wf::color_t{0.9, 0.9, 0.9, 1} /* fg_color */);
         set_text_params(params);
         // TODO: make the text smaller and display a useful message instead of a big explosion.
-        set_text("💥");
+        set_text(text);
         auto layer_node = output->node_for_layer(wf::scene::layer::LOCK);
-        wf::scene::add_back(layer_node, shared_from_this());
+        if (!is_displayed)
```

Comment:
Do we need an additional bool variable for this? AFAICT all you want to know is whether the node has alraedy been added to the scenegraph. You could do that by checking whether its parent is NULL.

Reply:
No need, I didn't realize `parent()` can be used for this :)